File: python/tests/mock_env_loader.py
from src.abstract_env_conf_loader import EnvConfigLoader


class GithubMockEnvConfig(EnvConfigLoader):
    """
      Github environment aware config loader.
      Implements EnvConfigLoader in order to be injected into EnvConfig
    """

    def __init__(self, environment):
        super().__init__(environment)

    def mock_set_categories(self, categories):
        self.__categories = categories

    def mock_set_data(self, data):
        self.__data = data

    def list_categories(self):
        return self.__categories

    def load(self, category):
        try:
            return self.__data[category.upper()]

        except Exception as ex:
            print(
                f'Failed loading and parsing config json content from branch/env "{self._env}"\nexception: {ex}'
            )
            return {}

File: ruby/tests/mock_env_loader.rb
require_relative '../src/abstract_env_config_loader'


# Config environment aware config loader - mock.
# Implements EnvConfigLoader in order to be injected into EnvConfig
class GithubMockEnvConfig < EnvConfigLoader
  def initialize(environment)
    super(environment)
  end

  def mock_set_categories(categories)
    @__categories = categories
  end

  def mock_set_data(data)
    @__data = data
  end

  def list_categories
    return @__categories
  end
  
  def load(category)
    begin
      return @__data[category.upcase]
    rescue StandardError => e
      puts "Failed loading and parsing config json content from branch/env '{@__environment}'\nexception: #{ex}"
    end
    
    return {}
  end
end
